cargo-near/src/util/tracking.rs

```diff
@@ -0,0 +1,84 @@
+use reqwest::Client;
+use std::{env, process::Command, str};
+use tracing::debug;
+
+#[derive(Debug, serde::Serialize)]
+struct MixpanelProperties {
+    token: String,
+    cli_version: String,
+    os: String,
+}
+
+#[derive(Debug, serde::Serialize)]
+struct TrackingData {
+    event: String,
+    properties: MixpanelProperties,
+}
+
+fn trim_newline(s: String) -> String {
+    let mut result = s;
+
+    if result.ends_with('\n') {
+        result.pop();
+        if result.ends_with('\r') {
+            result.pop();
+        }
+    }
+
+    result
+}
+
+pub(crate) fn track_usage() {
+    let output = if cfg!(target_os = "windows") {
+        Command::new("cmd")
+            .args(["/C", "near --version"])
+            .output()
+            .expect("failed to execute process")
+    } else {
+        Command::new("sh")
+            .arg("-c")
+            .arg("near --version")
+            .output()
+            .expect("failed to execute process")
+    };
+
+    let cli_version: String = match str::from_utf8(&output.stdout) {
+        Ok(v) => v.split(" ").collect::<Vec<_>>()[1].to_string(),
+        Err(_) => "unknown".to_string(),
+    };
+
+    let properties = MixpanelProperties {
+        token: "24177ef1ec09ffea5cb6f68909c66a61".to_string(),
+        cli_version: trim_newline(cli_version),
```

Comment:
Wow this is an overkill :-D

Cargo provides build-time env variables, and Cargo.toml version value is among them:

https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates

Reply:
And the request was to track CLI version

> I'd track CLI version instead of Rust version.

Reply:
@garikbesson Sorry for the confusion, `cargo-near` is also CLI and I meant "this [cargo-near] CLI program version"